CONTRIBUTING.md

```diff
@@ -0,0 +1,78 @@
+# New Developers
+## Coding Style
+We follow google's c++ coding style. See https://google.github.io/styleguide/cppguide.html and https://github.com/kokkos/kokkos-kernels/blob/master/.clang-format for details. 
+### Automate coding style via a pre-commit hook:
+```bash
+cat kokkos-kernels/.git/hooks/pre-commit
+for FILE in $(git diff --cached --name-only | egrep '.*\.cpp$|.*\.hpp$|.*\.h$')
+do
+  if [ -e $file ]; then
+    clang-format-8 -i -style=file $FILE
```

Comment:
Is `clang-format-8` is specific to MacOS? If so, some comments are needed here for Linux users. The SEMS modules for clang are putting `clang-format` in my PATH.

Reply:
In the end, it's only important that `clang-format` has version 8. Newer `clang-format` version indent slightly differently so that using a specific version is required.